api/writer/json.go

```diff
@@ -13,26 +12,32 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type JSONResponseWriter struct {
+type JSONOrYAMLResponseWriter struct {
```

Comment:
This really has nothing to do with JSONorYaml now.  Can you rename to `EncodingResponseWriter`